app/uk/gov/hmrc/emailverification/services/JourneyService.scala

```diff
@@ -86,20 +84,38 @@ class JourneyService @Inject() (
     }
   }
 
+  def lockIfNewEmailAddressExceedsCount(credId: String, emailAddress: String): Future[Boolean] = {
```

Comment:
Suggest you have boolean function isEmailAddressLocked(credId, emailAddress) that then calls verificationStatusRepository.lock(credId, emailAddress) if true.